apps/settings/js/modules/bluetooth/bluetooth_context.js

```diff
@@ -0,0 +1,852 @@
+      this.debug('_onAdapterDevicepaired evt = ' + evt);
+      // have to get device object in this event handler Ex. evt --> device
+
+      // Instead of adding the paired device in paired devices list,
+      // get paired devices from adapter directly.
+      this._refreshPairedDevicesInfo(adapter);
+    },
+
+    /**
+     * 'ondeviceunpaired' event handler from default adapter for updating
+     * unpaired device in remote/paired devices list.
+     *
+     * @access private
+     * @memberOf BluetoothContext
+     * @param {Object BluetoothAdapter} adapter
+     * @param {event} evt
+     */
+    _onAdapterDeviceunpaired:
+    function btc__onAdapterDeviceunpaired(adapter, evt) {
+      this.debug('_onAdapterDeviceunpaired evt = ' + evt);
+      // have to get the device address in this event handler Ex. evt -> address
+
+      // Instead of removing the paired device from paired devices list,
+      // get paired devices from adapter directly.
+      this._refreshPairedDevicesInfo(adapter);
+    },
+
+    /**
+     * 'defaultAdapter' change event handler from adapter manager for
+     * watch/unwatch default adapter relative event immediately.
+     *
+     * @access private
+     * @memberOf BluetoothContext
+     * @param {Object BluetoothAdapter} newAdapter
+     * @param {Object BluetoothAdapter} oldAdapter
+     */
+    _onDefaultAdapterChanged:
+    function btc__onDefaultAdapterChanged(newAdapter, oldAdapter) {
+      this.debug('_onDefaultAdapterChanged(): newAdapter = ' + newAdapter);
+      this.debug('_onDefaultAdapterChanged(): oldAdapter = ' + oldAdapter);
+
+      // save default adapter
+      this._defaultAdapter = newAdapter;
+
+      if (oldAdapter) {
+        // unwatch event since the old adapter is no longer usefull
+        this._unwatchDefaultAdapterOnattributechanged(oldAdapter);
+        this._unwatchDefaultAdapterOndeviceunpaired(oldAdapter);
+        this._unwatchDefaultAdapterOndevicepaired(oldAdapter);
+      }
+
+      if (newAdapter) {
+        // watch event since the new adapter is ready to access
+        this._initProperties(newAdapter);
+        this._watchDefaultAdapterOnattributechanged(newAdapter);
+        this._watchDefaultAdapterOndevicepaired(newAdapter);
+        this._watchDefaultAdapterOndeviceunpaired(newAdapter);
+      } else {
+        // reset properties only
+        this._resetProperties();
+      }
+    },
+
+    /**
+     * State of init for paired devices info. We only init paired devices info
+     * one time while _init() or 'onattributechanged' event comimg from default
+     * adapter.
+     *
+     * @access private
+     * @memberOf BluetoothContext
+     * @type {Boolean}
+     */
+    _hasInitPairedDevicesInfo: false,
+
+    /**
+     * Init paired devices information.
+     *
+     * @access private
+     * @memberOf BluetoothContext
+     * @param {Object} adapter
+     */
+    _initPairedDevicesInfo: function btc__initPairedDevicesInfo(adapter) {
```

Comment:
What is the purpose of this function? Can we simply call to `_refreshPairedDevicesInfo`?